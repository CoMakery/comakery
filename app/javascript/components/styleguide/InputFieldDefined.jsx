import React from 'react'
import classNames from 'classnames'
import InputField from './InputField'

class InputFieldDefined extends React.Component {
  render() {
    const {
      className,
      ...other
    } = this.props

    const classnames = classNames(
      'input-field__defined',
      className
    )

    return (
      <React.Fragment>
        <InputField
          className={classnames}
          type="defined"
          symbolLimit={0}
          {...other}
        />
      </React.Fragment>
    )
  }
}

export default InputFieldDefined
