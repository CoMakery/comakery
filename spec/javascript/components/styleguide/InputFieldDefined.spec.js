import React from 'react'
import { shallow, mount, render } from 'enzyme'
import InputFieldDefined from 'components/styleguide/InputFieldDefined'

describe('InputFieldDefined', () => {
    it('renders correctly without props', () => {
      const wrapper = shallow(<InputFieldDefined/>)

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('.input-field__defined').props().type).toBe('defined')
      expect(wrapper.exists('.input-field--title--counter')).not.toBe()
    })

    it('renders correctly with custom className', () => {
      const wrapper = shallow(<InputFieldDefined className='__test' />)

      expect(wrapper.exists('.input-field__defined.__test')).toBe(true)
    })
})
