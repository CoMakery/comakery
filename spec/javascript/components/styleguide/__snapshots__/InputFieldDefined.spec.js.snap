// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFieldDefined renders correctly without props 1`] = `
<Fragment>
  <InputField
    checkboxText=""
    checked={false}
    className="input-field__defined"
    disabled={false}
    errorText=""
    eventHandler={[Function]}
    imgInputRef={null}
    imgPreviewDimensions="40x40"
    imgPreviewUrl=""
    imgRequirements=""
    imgVerifier={[Function]}
    name="field"
    pattern=".*"
    placeholder="Please enter value"
    readOnly={false}
    required={false}
    selectEntries={Array []}
    symbolLimit={0}
    title="title"
    type="defined"
    value=""
  />
</Fragment>
`;
